scripts/acceptance-test/src/bin/setup.rs

```diff
@@ -0,0 +1,435 @@
+use stf_starter::sov_modules_api::transaction::{
+    PriorityFeeBips, Transaction, UnsignedTransaction,
+};
+use stf_starter::sov_modules_api::{CryptoSpec, RawTx};
+use stf_starter::RuntimeCall;
+use tokio_stream::StreamExt;
+
+use tracing::info;
+
+fn compare_tx_info_and_accepted_tx(
+    tx_info: &types::TxInfoWithConfirmation,
+    accepted_tx: &types::ApiAcceptedTx,
+    description: &str,
+) {
+    // Compare shared fields
+    assert_eq!(
+        tx_info.events, accepted_tx.events,
+        "{}: events should match",
+        description
+    );
+    assert_eq!(
+        tx_info.id, accepted_tx.id,
+        "{}: id should match",
+        description
+    );
+    assert_eq!(
+        tx_info.tx_number,
+        Some(accepted_tx.tx_number),
+        "{}: tx_number should match",
+        description
+    );
+
+    // TxInfoWithConfirmation has receipt wrapped in Option, ApiAcceptedTx has it directly
+    if let Some(ref receipt) = tx_info.receipt {
+        assert_eq!(
+            receipt, &accepted_tx.receipt,
+            "{}: receipt should match",
+            description
+        );
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // Initialize tracing subscriber with RUST_LOG environment variable, fallback to info
+    tracing_subscriber::fmt()
+        .with_env_filter(
+            tracing_subscriber::EnvFilter::try_from_default_env()
+                .unwrap_or_else(|_| tracing_subscriber::EnvFilter::new("info")),
+        )
+        .init();
+
+    let directories = Directories::new()?;
+    let password = generate_postgres_password()?;
+    start_and_wait_for_postgres_ready(POSTGRES_CONTAINER_NAME, &password)?;
+    interpolate_config(&password, &directories)?;
+
+    info!(
+        "Starting rollup from rollup workspace root: {}",
+        directories.rollup_root.display()
+    );
+    let rollup = Command::new("cargo")
+        .args([
+            "run",
+            "--release",
+            "--",
+            "--rollup-config-path",
+            &directories
+                .output_dir
+                .join("config.toml")
+                .display()
+                .to_string(),
+            "--genesis-path",
+            &directories
+                .acceptance_test_dir
+                .join("genesis.json")
+                .display()
+                .to_string(),
+            "--stop-at-rollup-height",
+            &(NUM_SOAK_BATCHES + 10).to_string(),
+        ])
+        .current_dir(directories.rollup_root.clone())
+        .env("RUST_LOG", "info")
+        .stdout(std::fs::File::create(
+            directories.output_dir.join("rollup.log"),
+        )?)
+        .spawn()
+        .expect("Failed to start rollup");
+
+    // First, run some manual setup. This creates and checks some very simple state with expensive consistency checks.
+    do_manual_setup(directories.clone()).await?;
+    let throughput_report = run_soak(directories.clone(), rollup, 3, true).await?;
```

Comment:
Would it be worthwhile to re-run another manual setup (with similar finegrained assertions) once more after the soak run, i.e. on top of a big existing rollup state rather than on a brand new clean one?

Reply:
I think at that poitn we'll have hit diminishing returns. The manual setup is really just a 1-time sanity check that (for example) transaction numbering doesn't always return 0 etc.